src/line_shapes/get_line_shapes.jl

```diff
@@ -0,0 +1,128 @@
+    ll .*= calc_doppler_factor(RV_estimate)
+    #get array dimensions
+    n_lines = length(ll)
+    n_depths = length(frac_depths)
+    #allocate memory for line shapes - could be optimized by referencing memory in get_line_shapes(liine_list,clt)
+    bisectors = zeros(n_depths,n_lines)
+    widths = zeros(n_depths,n_lines)
+    depths = zeros(n_lines)
+    valid_lines = ones(Bool,n_lines)
+    #pick which chunk to measure each line shape in
+    line_chunk_index = zeros(Int64,n_lines)
+    for i in 1:n_lines
+        line_chunk_index[i] = findmin(abs.((chunk_centers .- ll[i]) ./ chunk_centers))[2] # TODO: change line_chunk_index to pick higher SNR of valid chunks
+    end
+    #measure the bisectors and widths
+    for i in 1:n_lines
+        #find pixels within v_width of ll[i]
+        line_high = ll[i] * calc_doppler_factor(v_width)
+        line_low = ll[i] * calc_doppler_factor(-v_width)
+        index_low = searchsortedfirst(chunk_list.data[line_chunk_index[i]].λ,line_low) - 1
+        index_high = searchsortedfirst(chunk_list.data[line_chunk_index[i]].λ,line_high)
+        if (index_low < 1) | (index_high > length(chunk_list.data[line_chunk_index[i]].λ))
+            if verbose
+                println("Warning: line number ",i," is not contained in chunk_list.")
+            end
+            valid_lines[i] = false
+        else
+            #get λs and fluxes for these pixels
+            local_λ = view(chunk_list.data[line_chunk_index[i]].λ,index_low:index_high)
+            local_flux = view(chunk_list.data[line_chunk_index[i]].flux,index_low:index_high)
+            #calculate the depth by taking the minimum of the middle 50% of the line's pixels #TODO: add make this fraction an argument as in calc_line_width()
+            depths[i] = 1.0 - minimum(local_flux[ Int(floor(length(local_flux) / 4)) : Int(ceil(length(local_flux) * 3 / 4)) ]) / maximum(local_flux)
+            for j in 1:n_depths
+                #calculate the line widths and bisectors # TODO opt: could merge calc_line_bis with calc_line width for performance
+                b = RvSpectMLBase.calc_line_bisector_at_frac_depth(local_λ, local_flux, frac_depth=frac_depths[j])
+                w = RvSpectMLBase.calc_line_width(local_λ, local_flux, frac_depth=frac_depths[j])
+                @assert j != 1 || (!isnan(b) && !isnan(w)) #line minumum is wrong or line doesn't exist
+                # in the case of NaNs, use the previous depth's value (this is why depths must be sorted decreasing)
+                if isnan(b)
+                    bisectors[j,i] = bisectors[j-1,i]
+                else
+                    bisectors[j,i] = b
+                end
+                if isnan(w)
+                    widths[j,i] = widths[j-1,i] #TODO: figure out what to do here to make more sense
+                else
+                    widths[j,i] = w
+                end #end "if isnan(w)"
+            end #end "for j in 1:n_depths"
+        end #end if line is not in chunk_list
```

Comment:
Not needed for this pull request, but in the future, you might consider making this into a funciton that opperates on a single line.  
If you make it a mutating function, then it could be called here to avoid code duplication.